### PR TITLE
Setup azure0 bridge when starting kubelet

### PR DESCRIPTION
**Pull request by zhedahht**
We have a bug that azure0 bridge is not initialized as excepted in some cases. We setup the bridge if it's not initialize before when starting kubelet, as suggested by partner teams.

### CODE REVIEW

File: parts/linux/cloud-init/artifacts/configure_azure0.sh [hunk 1]
```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env bash
+
+# Check if the azure cni config is there... no need to run this script if not
+# Also don't want to run this when not using azure-cni
+[ ! -f /etc/cni/net.d/10-azure.conflist ] && exit 0
+
+# Check if the azure0 bridge is already configured
+# We don't need to run if so.
+ip link show azure0 && exit 0
+
```
Comment (xuto2) [inline parts/linux/cloud-init/artifacts/configure_azure0.sh:10]: CNI team mentions that this is not needed for calico network policy, so we should check the network policy and skip this configuration. The network policy is a parameter in cse_cmd.sh, we can get the value in the similar way

**Comment by xuto2** [comment]
to test, create a tag on your private branch and use it to vendor to AKS RP and you can test in standalone env or in E2E.

File: parts/linux/cloud-init/artifacts/configure_azure0.sh (hunk 1 above)
Comment (zhedahht) [reply parts/linux/cloud-init/artifacts/configure_azure0.sh:16]: Tried to use "{{GetParameter "networkPolicy"}}" in nodecustomdata.yml. Is it the right way to pass the network policy as a parameter?

File: pkg/agent/testdata/AKSUbuntu1604+DynamicKubelet/line43.sh [hunk 2]
```diff
@@ -26,6 +26,10 @@ ExecStartPre=/sbin/sysctl -w net.ipv4.neigh.default.gc_thresh3=16384
 
 ExecStartPre=-/sbin/ebtables -t nat --list
 ExecStartPre=-/sbin/iptables -t nat --numeric --list
+
+
+ExecStartPre=ExecStartPre=/usr/local/bin/configure_azure0.sh
```
Comment (xuto2) [inline pkg/agent/testdata/AKSUbuntu1604+DynamicKubelet/line43.sh:31]: just one ExecStartPre=

File: parts/linux/cloud-init/artifacts/configure_azure0.sh [hunk 3]
```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env bash
+
+# Check if the azure cni config is there... no need to run this script if not
+# Also don't want to run this when not using azure-cni
+[ ! -f /etc/cni/net.d/10-azure.conflist ] && exit 0
+
+# CNI team mentions that this is not needed for calico network policy to run this script
+if [[ "${NETWORK_POLICY}" == "calico" ]]; then
```
Comment (xuto2) [inline parts/linux/cloud-init/artifacts/configure_azure0.sh:8]: if [[ "{{GetParameter "networkPolicy"}}" == "calico" ]]; then

File: parts/linux/cloud-init/nodecustomdata.yml [hunk 4]
```diff
@@ -74,6 +74,14 @@ write_files:
     {{GetVariableProperty "cloudInitData" "reconcilePrivateHostsService"}}
 {{- end}}
 
+- path: /usr/local/bin/configure_azure0.sh
+  permissions: "0544"
+  encoding: gzip
+  owner: root
+  content: !!binary |
+    {{GetVariableProperty "cloudInitData" "configureAzure0Script"}}
+    NETWORK_POLICY={{GetParameter "networkPolicy"}}
```
Comment (xuto2) [inline parts/linux/cloud-init/nodecustomdata.yml:83]: remove this

File: pkg/agent/testdata/AKSUbuntu1604+DynamicKubelet/CustomData [hunk 5]
```diff
@@ -35,12 +35,20 @@ write_files:
 
 
 
+- path: /usr/local/bin/configure_azure0.sh
+  permissions: "0544"
+  encoding: gzip
+  owner: root
+  content: !!binary |
+
```
Comment (xuto2) [inline pkg/agent/testdata/AKSUbuntu1604+DynamicKubelet/CustomData:43]: this should not be empty

File: parts/linux/cloud-init/artifacts/kubelet.service [hunk 6]
```diff
@@ -28,6 +28,10 @@ ExecStartPre=/sbin/sysctl -w net.ipv4.neigh.default.gc_thresh3=16384
 
 ExecStartPre=-/sbin/ebtables -t nat --list
 ExecStartPre=-/sbin/iptables -t nat --numeric --list
+
```
Comment (xuto2) [inline parts/linux/cloud-init/artifacts/kubelet.service:31]: remvoe teh empty line

File: pkg/agent/testdata/AKSUbuntu1604+DynamicKubelet/line43.sh (hunk 2 above)
Comment (zhedahht) [reply pkg/agent/testdata/AKSUbuntu1604+DynamicKubelet/line43.sh]: fixed.

File: parts/linux/cloud-init/artifacts/configure_azure0.sh (hunk 3 above)
Comment (xuto2) [reply parts/linux/cloud-init/artifacts/configure_azure0.sh:8]: sorry , GetParameter won't work currently. it needs some refactoring. But I have one workaround. see my other comment

File: parts/linux/cloud-init/nodecustomdata.yml (hunk 4 above)
Comment (xuto2) [reply parts/linux/cloud-init/nodecustomdata.yml:83]: actually moving this  line to between L382 and L395 should work

File: parts/linux/cloud-init/nodecustomdata.yml [hunk 7]
```diff
@@ -74,6 +74,13 @@ write_files:
     {{GetVariableProperty "cloudInitData" "reconcilePrivateHostsService"}}
 {{- end}}
 
+- path: /usr/local/bin/configure_azure0.sh
+  permissions: "0544"
```
Comment (xuto2) [inline parts/linux/cloud-init/nodecustomdata.yml:78]: nit: 0744?

File: pkg/agent/variables.go [hunk 8]
```diff
@@ -29,6 +29,7 @@ func getCustomDataVariables(config *NodeBootstrappingConfiguration) paramsMap {
 	}
 
 	cloudInitData := cloudInitFiles["cloudInitData"].(paramsMap)
+	cloudInitData["configureAzure0Script"] = getBase64EncodedGzippedCustomScript(kubernetesConfigAzure0Script, config)
```
Comment (xuto2) [inline pkg/agent/variables.go:32]: move to L28

File: parts/linux/cloud-init/nodecustomdata.yml (hunk 7 above)
Comment (zhedahht) [reply parts/linux/cloud-init/nodecustomdata.yml:78]: I took health-monitor.sh as an example, which was 0544.

File: pkg/agent/variables.go (hunk 8 above)
Comment (zhedahht) [reply pkg/agent/variables.go]: Moved. Thanks

**Comment by xuto2** [comment]
lgtm.

**Pull request merged**